src/Benchmarks/SingleQueryEfMiddleware.cs

```diff
@@ -35,10 +35,8 @@ public SingleQueryEfMiddleware(RequestDelegate next)
             {
                 var db = (ApplicationDbContext)httpContext.RequestServices.GetService(typeof(ApplicationDbContext));
 
-                db.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;
-
                 var id = _random.Next(1, 10001);
-                var row = await db.World.FirstAsync(w => w.Id == id);
+                var row = await db.World.AsNoTracking().FirstAsync(w => w.Id == id);
```

Comment:
I am not sure there is currently a way to have the query expression created only once and stashed in a static field because the DbContext instance is used to bootstrap the process. 

FWIW if we thought many customers would be ok with the ugly code required to stash the expression in a static field just to avoid the overhead of re-creating the expression on every request we could as well consider (re)adding the `CompiledQuery` feature to the backlog, since that would also improve the query translation lookup.